### ISSUE TITLE
Review low priority alarms and create issues/resolve recurring ones

### ISSUE THREAD

**Issue by davidkelliott** [ministryofjustice/modernisation-platform#8369]
### User Story

As a MP engineer
I want only valid alarms to appear in the #modernisation-platform-low-priority-alarms channel
So that I can respond to things that I need to respond to

### Value / Purpose

We want to review the low priority alarms as part of the milk monitor.  Before we start doing this we should ensure that any common repeating false alarms are resolved. The value in this issue is having a low priority alarm channel that only has alerts in it which we need to respond to or verify that the action which caused the alarm is not malicious.

### Context / Background

There are some alarms which regularly go off such as unauthorised API requests, not due to malicious activity, but due to misconfiguration of a service or policy trying to access something it shouldn't, or not being allow access to something it should have.

This issue is to go back through the last 2 weeks of alarms in the #modernisation-platform-low-priority-alarms channel and resolve (or raise issue if too complex) the underlying issue causing the alarm.

We want to be in a position where alarms only go off when there is action required by us.

### Definition of Done

- [x] #modernisation-platform-low-priority-alarms channel alarms reviewed over the last 2 weeks
- [x] false alarms resolved or issues created to resolve them

**Comment by mikereiddigital** [comment]
I'm looking at the unauthorised-api-calls metric in modernisation-platform as this gives us a more granular view of exactly when these exceptions are taking place. With this information, I am then pulling cloudtrail events for 1 hour slots out as json & then searching for the specific event codes as this cannot be used via the console.

**Comment by mikereiddigital** [comment]
Given the numerous ThrottlingException issues I was encountering with using the CLI, I wrote a python script & a bash wrapper that:

- For a given period (e.g. two weeks of 12 hours, gets cloudtrail data in small chunks of one hour each.
- Iterates through the data and gets events that match a list of specific error codes that relate to the unauthorised-api-calls metric.
- Outputs the list as pseudo json (correct formatting still to do).

With this I can get a whole two weeks of cloudtrail errors for an account. Starting with modernisation-platform.

**Comment by mikereiddigital** [comment]
First bug raised - https://github.com/ministryofjustice/modernisation-platform/issues/8545

**Comment by mikereiddigital** [comment]
Also https://github.com/ministryofjustice/modernisation-platform/issues/8546

**Comment by mikereiddigital** [comment]
Raised bug #8558

**Comment by mikereiddigital** [comment]
Also bug #8559

**Comment by mikereiddigital** [comment]
Have been told that the cloudtrail event data is available in Cloudwatch (log group name Cloudtrail) and accessing is should be much faster that polling cloudtrail directly. Will look at a script that can create a log insights query on the fly for a given period & will see how such a query can be used in the Observability platform to show these as and when they occur.

**Comment by mikereiddigital** [comment]
Have added a new Grafana report to the Observability Platform -[Cloudtrail - AccessDenied Errors ](https://g-9d213fbc19.grafana-workspace.eu-west-2.amazonaws.com/d/de57uk6q0osu8b/cloudtrail-accessdenied-errors?orgId=1). This makes searching across accounts for these errors far easier than running python scripts. The only to-do is to add a Data Source for Modernisation Platform account and add that to the query variable.

**Comment by SimonPPledger** [comment]
looks good to me. Moving to done
